tests/lib/rules/no-extra-parens.js

```diff
@@ -152,7 +154,6 @@ ruleTester.run("no-extra-parens", rule, {
         "var o = { foo: (function() { return bar(); })() };",
         "o.foo = (function(){ return bar(); })();",
         "(function(){ return bar(); })(), (function(){ return bar(); })()",
-        "((function(){ return bar(); })())",
```

Comment:
We need to be very careful about deciding valid cases are no longer valid. Can you check when this case was added (use blame).